apps/api/src/app/import/import.service.ts

```diff
@@ -322,13 +336,70 @@ export class ImportService {
           value,
           currency,
           userCurrency
-        )
+        ),
+        isDuplicate
       });
     }
-
+    console.log(activities);
     return activities;
   }
 
+  private async labelDuplicateActivities({
+    activitiesDto,
+    userId
+  }: {
+    activitiesDto: Partial<CreateOrderDto>[];
+    userId: string;
+  }): Promise<Partial<OrderDto>[]> {
+    const existingActivities = await this.orderService.orders({
+      include: { SymbolProfile: true },
+      orderBy: { date: 'desc' },
+      where: { userId }
+    });
+
+    const activitiesDtoWithDuplication: Partial<OrderDto>[] = [];
```

Comment:
Can you use `Partial<Activity>[]`? Then you could get rid of `OrderDto` completely

Reply:
I have used `Partial<Activity>[]` but I think we will get a type error in the loop, because Activity doesn't contain `currency`, and some other fields maybe? Or Am I missing something?

![image](https://user-images.githubusercontent.com/64573465/236678957-f72cee74-1536-4c90-91f2-e61511372e06.png)

I can clean up `OrderDto` like this if you prefer

![image](https://user-images.githubusercontent.com/64573465/236679038-30c207c8-8e86-4aac-ae4d-7c05cc86af4e.png)

Reply:
I can clean up data definition like this

![image](https://user-images.githubusercontent.com/64573465/236681106-521aca1b-3ae5-4380-b007-4631e278d0b2.png)

Reply:
You are right. Extending is fine. I would remove `@IsBoolean()` as we do not validate here.